### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by rjpals**
This adds the ability to use TypeScript and converts a single file (`src/shell/notifications.js`) to TS. 

With this configuration, TS is entirely optional. The source tree consist of TS and JS interoperating with each other and developers will be able to freely choose between them (for now.) This will allow us to incrementally adopt TS without an all-out rewrite.

For information, see [the spec](https://docs.google.com/document/d/1e_ALIynxoEWcxB4pb6zl_tqzUKIpnxjS4oag2BAhScI)

### CODE REVIEW

File: src/shell/store/notifications.ts [hunk 1]
```diff
@@ -1,4 +1,10 @@
-export function notifications(state = [], action) {
+type Action = {
```
Comment (agalin920) [inline src/shell/store/notifications.ts:1]: Are we gonna go with 'type' over 'interface' ?

File: src/shell/store/notifications.ts [hunk 2]
```diff
@@ -8,15 +14,21 @@ export function notifications(state = [], action) {
...
+        return notification;
       });
 
     default:
       return state;
   }
 }
 
-export function notify(data) {
+type Data = {
```
Comment (agalin920) [inline src/shell/store/notifications.ts:25]: I think Data is too arbitrary and not indicative enough of what this is. Shouldn't we should name this Notification? Especially in the case we ever need to export this type to a component

**Review by agalin920 — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -101,6 +101,7 @@
     "@babel/plugin-transform-runtime": "^7.10.4",
     "@babel/preset-env": "^7.14.7",
     "@babel/preset-react": "^7.14.5",
+    "@babel/preset-typescript": "^7.18.6",
```
Comment (agalin920) [inline package.json:104]: Is this needed?

File: src/shell/store/notifications.ts [hunk 4]
```diff
@@ -1,4 +1,10 @@
-export function notifications(state = [], action) {
+type Action = {
+  type: "NEW_NOTIFICATION" | "REMOVE_NOTIFICATION";
+  data: Data;
+  epoch: Date;
+};
+
+export function notifications(state = [], action: Action) {
```
Comment (agalin920) [inline src/shell/store/notifications.ts:7]: State should be defined

File: src/shell/webpack.config.js [hunk 5]
```diff
@@ -74,8 +74,10 @@ module.exports = async (env) => {
         utility: path.resolve(__dirname, "../utility"),
         apps: path.resolve(__dirname, "../apps"),
       },
+      extensions: [".ts", ".js"],
     },
     plugins: [
+      /*
```
Comment (shrunyan) [inline src/shell/webpack.config.js:80]: why has this been commented out?

**Review by shrunyan — APPROVED** [review]

**Pull request merged**